File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FSwitch.java
package io.github.gofaith.faithdroid.FViews;

import android.graphics.Color;
import android.support.v7.widget.SwitchCompat;
import android.view.View;
import android.widget.CompoundButton;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FSwitch extends FView implements AttrSettable, AttrGettable {
    public SwitchCompat v;

    public FSwitch(UIController controller) {
        parentController=controller;
        v = new SwitchCompat(parentController.activity);
        view=v;
    }
    @Override
    public String getAttr(String attr) {
        String str = getUniversalAttr(attr);
        if (str != null) {
            return str;
        }
        switch (attr){
            // ------------------------------------------
            case "Checked":
                return String.valueOf(v.isChecked());
            case "Enabled":
                return String.valueOf(v.isEnabled());
            case "TextOn":
                return v.getTextOn().toString();
            case "TextOff":
                return v.getTextOff().toString();
        }
        return "";
    }

    @Override
    public void setAttr(String attr, final String value) {
        if (setUniversalAttr(attr, value)) {
            return;
        }
        switch (attr){
            // -------------------------------------------------------------------
            case "TextColor":
                try {
                    v.setTextColor(Color.parseColor(value));
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "TextSize":
                try {
                    v.setTextSize(dp2pixel(parentController.activity,Float.valueOf(value)));
                } catch (Exception e) {
                    e.printStackTrace();
                    return;
                }
                break;
            case "Enabled":
                if (value.equals("true")) {
                    v.setEnabled(true);
                } else {
                    v.setEnabled(false);
                }
                break;
            case "TextOn":
                v.setShowText(true);
                v.setTextOn(value);
                break;
            case "TextOff":
                v.setShowText(true);
                v.setTextOff(value);
                break;
            case "OnCheckedChangeListener":
                v.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                    @Override
                    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                        Faithdroid.triggerEventHandler(value, String.valueOf(isChecked));
                    }
                });
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FSwitch.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.graphics.Color
import android.widget.CompoundButton
import androidx.appcompat.widget.SwitchCompat
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.Toolkit.dp2pixel
import io.github.stevenzack.faithdroidapp.UI.UIController


class FSwitch(controller: UIController) : FView(), AttrSettable, AttrGettable {
    var v: SwitchCompat

    init {
        parentController = controller
        v = SwitchCompat(parentController!!.activity)
        view = v
    }

  override  fun getAttr(attr: String): String {
        val str = getUniversalAttr(attr)
        if (str != null) {
            return str
        }
        when (attr) {
            // ------------------------------------------
            "Checked" -> return v.isChecked().toString()
            "Enabled" -> return v.isEnabled().toString()
            "TextOn" -> return v.getTextOn().toString()
            "TextOff" -> return v.getTextOff().toString()
        }
        return ""
    }

  override  fun setAttr(attr: String, value: String) {
        if (setUniversalAttr(attr, value)) {
            return
        }
        when (attr) {
            // -------------------------------------------------------------------
            "TextColor" -> try {
                v.setTextColor(Color.parseColor(value))
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "TextSize" -> try {
                v.setTextSize(dp2pixel(parentController!!.activity, java.lang.Float.valueOf(value)))
            } catch (e: Exception) {
                e.printStackTrace()
                return
            }

            "Enabled" -> if (value == "true") {
                v.setEnabled(true)
            } else {
                v.setEnabled(false)
            }
            "TextOn" -> {
                v.setShowText(true)
                v.setTextOn(value)
            }
            "TextOff" -> {
                v.setShowText(true)
                v.setTextOff(value)
            }
            "OnCheckedChangeListener" -> v.setOnCheckedChangeListener(CompoundButton.OnCheckedChangeListener { buttonView, isChecked ->
                Faithdroid.triggerEventHandler(
                    value,
                    isChecked.toString()
                )
            })
        }
    }
}
